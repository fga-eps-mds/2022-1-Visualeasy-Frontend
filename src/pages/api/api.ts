import axios from "axios";

axios.defaults.baseURL = process.env.NEXT_PUBLIC_BASE_URL;
export interface Variavel {
    id?: number,
    variavel?: string;
    data?: Date;
    valor?: number;
}
export interface Variavel extends Array<Variavel> { }

const paramstype = {
    variavel: "Bulbassauro",
    startDate: '2022-06-30T06:18:50',
    endDate: '2022-06-30T06:26:14'
}
export async function listaNomeVariaveis() {
    try {
        const resposta = await axios.get("variavel/");
        return resposta.data;
    } catch (e) {
        return e
    }

};

export const postAllData = async (params: any, dataapi: any) => {
    try {

        const { data }: any = await axios.post(`/variavel/${params}`, dataapi).catch(function (error) {
            console.error(error);
        });
        return data

    } catch (e) {
        return e
    }
}

