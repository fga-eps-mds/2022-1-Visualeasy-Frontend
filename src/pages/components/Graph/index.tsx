import React from 'react';
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,

} from 'chart.js';

import { Line } from 'react-chartjs-2';

import {Box} from "@chakra-ui/react"
ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);
function getRandomColor() {
  var letters = '0123456789ABCDEF';
  var color = '#';
  for (var i = 0; i < 6; i++) {
    color += letters[Math.floor(Math.random() * 16)];
  }
  return color;
}

export default function Graph(dataBase) {
   const data = {
    datasets: [
      {
        label: 'Dataset 1',
        data: dataBase.dataBase.variavels,
        lineTension: 0.5,  
        backgroundColor: `${getRandomColor()}`,
      },

    ],
  };
  const options = {
    type:"line",
   
    bezierCurve : false,
      parsing: {
        xAxisKey: 'data',
        yAxisKey: 'valor'
      },
      elements: {
        line: {
            tension: 0
        }
    },
  }
    return (
      <Box height="400px" w="100%">
      <Line data={data} options={options} />
    </Box>
    );
  };