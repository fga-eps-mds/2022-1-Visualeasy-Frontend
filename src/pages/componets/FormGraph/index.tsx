import { 
    VStack,
    Input,
    Text,
    Box,
    HStack,
    UnorderedList,
    ListItem,
    Tag,
    TagLabel,
    TagCloseButton,
    
} from "@chakra-ui/react";

import {
    AsyncCreatableSelect,
    AsyncSelect,
    ChakraStylesConfig,
    CreatableSelect,
    Select
  } from "chakra-react-select";

const variables = [
    {
        label: "Charmander",
        value: "charmander",
        id: "charmander",
        variant: "outline"
    },
    {
        label: "Squirtle",
        value: "squirtle",
        id: "squirtle"
    },
    {
        label: "Bulbassauro",
        value: "bulbassauro",
        id: "bulbassauro"
    },
]

const selectStyles: ChakraStylesConfig = {
    container: (provided, state) => ({
        ...provided,
        p: 0,
        w: "100%"
    }),

    placeholder: (provided, state) => ({
        ...provided,
        fontSize: "12px"
    })
}

export default function FormGraph() {

    return (
        <VStack w='100%'>
            <Input type="text" id={"title"} placeholder="Gráfico 1" value={""} />
            
            <Select
                name="variables"
                placeholder="Selecione as variáveis"
                isMulti
                id="variables-select"
                closeMenuOnSelect={false}
                size="md"
                tagVariant="solid"
                chakraStyles={selectStyles}
                options={variables}
            />

        </VStack>
    );
};
