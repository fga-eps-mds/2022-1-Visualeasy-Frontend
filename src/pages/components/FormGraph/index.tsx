import { VStack, Input, Button } from "@chakra-ui/react";
import { ChakraStylesConfig, Select } from "chakra-react-select";
import { useState, useEffect } from "react";
import { listaNomeVariaveis } from "../../api/api";
import RatioSelect from "../RatioTimeChart/Index";

const selectStyles: ChakraStylesConfig = {
    container: (provided) => ({
        ...provided,
        p: 0,
        w: "100%"
    }),

    placeholder: (provided) => ({
        ...provided,
        fontSize: "12px"
    })
}

export default function FormGraph({ FormGraphProps, disablebutton }: any) {
    const [GraphParameters, setGraphParameters] = useState('')
    const [variablesName, setVariablesName] = useState([{}])
    const [graphName, setgraphName] = useState("Gr치fico")
    const [variavelSelect, setvariavelSelect] = useState([])


    useEffect(() => {
        getListaNomes()
    }, []);
    var newArr = variavelSelect.map(function (val, index) {
        return (val.value)
    })

    const getListaNomes = async () => {
        const { variavels } = await listaNomeVariaveis();
        let variablesName = []

        for (let i = 0; i < variavels.length; i++) {
            let option = {
                label: variavels[i].variavel,
                value: variavels[i].variavel,
                id: variavels[i].variavel,
            }
            variablesName.push(option)
        }
        variablesName[0] = { ...variablesName[0], variant: "outline" }
        setVariablesName(variablesName)
    }

    function Gerar() {
        const variavel = [] = Object.keys(newArr)
            .map(function (key) {
                return newArr[key];
            });

        const FormGraphData = {
            id: new Date().getTime(),
            ...GraphParameters,
            variavel,
            graphName
        };

        FormGraphProps({
            ...FormGraphData
        });

    }
    return (
        <VStack w='100%'>
            <Input type="text" id={"title"} placeholder="Gr치fico 1" onChange={(e) => { setgraphName(e.target.value) }} />

            <Select
                name="variables"
                placeholder="Selecione as vari치veis"
                isMulti
                id="1"
                instanceId={"1"}
                closeMenuOnSelect={false}
                size="md"
                tagVariant="solid"
                onChange={(e) => setvariavelSelect(e)}
                chakraStyles={selectStyles}
                options={variablesName}
            />
            <RatioSelect RatioRange={(e) => setGraphParameters(e)} />
            {disablebutton ?
                (<Button colorScheme='red' size='lg' onClick={() => Gerar()}>Gerar gr치fico</Button>
                ) : (
                    <></>
                )
            }

        </VStack>
    );
};

