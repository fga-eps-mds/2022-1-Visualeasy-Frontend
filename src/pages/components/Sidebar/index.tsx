import React, { useState } from "react";
import {

  Image,
  Box,

  Tabs,
  TabList,
  Tab,
  TabPanels,
  TabPanel,
  Center,

  Grid,
  GridItem,
  CloseButton
} from "@chakra-ui/react";


import FormGraph from "../FormGraph";
import FormGraphinfo from "../FormShowInfo";

import Graph from "../Graph";
interface EnumServiceGetOrderBy {
  [index: number]: string;
}

interface dataFormProps {
  intervalo: number
  variavel: EnumServiceGetOrderBy[]
  startDate?: string
  endDate?: string
}

export default function Sidebar({ SidebarData }: any) {
  const [dataForm, setDataForm] = useState<dataFormProps>({
    intervalo: 0,
    variavel: []
  })


  const [postList, setPostList] = useState([]);


  const addPostlist = (postitem) => setPostList([...postList, postitem]);


  function deletPostList(deletitem) {
    var filtered =  postList.filter((e)=>(e.id!==deletitem))
    setPostList(filtered)
   // setPostList(postList.filter((e) => { e.id !== deletitem }));
  }


  return (
    <Tabs isFitted defaultIndex={0} className="Tabs" variant='enclosed'>
      <Grid
        templateAreas={`"nav main"
                  "nav main"
                  "nav main"`}
        gridTemplateRows={'50px 1fr 30px'}
        gridTemplateColumns={'340px 1fr'}
        h="100vh"
        gap='1'
        color='blackAlpha.700'
        fontWeight='bold'

      >
        <GridItem pl='2' p={2} bg='#F6F6f6' area={'nav'}>
          <Box marginBottom="10px">
            <Center>
              <Image
                id="logo-retangular"
                src="images/logo-retangular.png"
                alt="Logo"
              />
            </Center>
          </Box >
          <TabList  mb="1em" overflowY="hidden" overflowX="auto" >
            <Tab>
              Gr√°fico 1
            </Tab>
            {
              postList.map((e, index) => (
                <Tab key={index}>
                  {e.graphName}
                  <CloseButton size='sm' onClick={() => { deletPostList(e.id) }} />
                </Tab>
              ))

            }
          </TabList>
          <TabPanels>
            <TabPanel>
              <FormGraph FormGraphProps={addPostlist} disablebutton />
            </TabPanel>
            {postList.map((e,index) => (
              <TabPanel key={index}>
                <FormGraphinfo getDataFrom={e} />
              </TabPanel>
            ))}
          </TabPanels>
        </GridItem>
        <TabPanels>

          <TabPanel>
            <GridItem pl='2' area={'main'}>
              <Graph dataBase={dataForm} />
            </GridItem>
          </TabPanel>
          {postList.map((e, index) => (
            <TabPanel key={index}>
              <GridItem pl='2' area={'main'}>
                <Graph dataBase={e} />
              </GridItem>
            </TabPanel>
          ))
          }

        </TabPanels>

      </Grid>
    </Tabs>
  );
};
