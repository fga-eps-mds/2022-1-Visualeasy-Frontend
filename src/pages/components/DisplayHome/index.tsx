import React, { useEffect, useState } from "react";
import axios from "axios";
import { Grid, GridItem, HStack } from "@chakra-ui/react";
import Sidebar from "../Sidebar";
import Graph from "../Graph";
import { api, listaNomeVariaveis,  } from "../../api/api";
// import Chart from "./componetes/Charts";
// import FooterChart from "./componetes/FooterCharts";

// var rows = [];
// for (var i = 0; i < numrows; i++) {
//     rows.push(ObjectRow());
// }
// return tbody(rows);

// let userData = {
//   variavel: "None"
//   startDate: '2022-06-30T06:18:50',
//   endDate: '2022-06-30T06:26:14'
// };
// var data = [];
// for (var i = 0; i < example["variavel"].length; i++) {
//   userData["variavel"] = example["variavel"][i]
//   const row  = await api.post('/filtered', userData)
//   data.push(row)
  
// }



   



    interface EnumServiceGetOrderBy {
      [index: number]: string;
  }
  interface dataFormProps {
    intervalo:number
    variavel:EnumServiceGetOrderBy[]
  }

export default function DisplayHome() {
  const [database, setDatabase] = useState([])
  const [dataForm, setDataForm] = useState<dataFormProps>({
    intervalo:0,
    variavel:[]
})

 useEffect(() => {
  async function getData(alldata=[]) {
     const userData = {
         variavel:await alldata["variavel"],
         // intervalo: 1
         startDate: '2022-06-30T06:18:50',
         endDate: '2022-06-30T06:26:14'
     };
     //const { data } = await api.post('/filteredByPeriod', userData)
      await api.post('/filtered', userData).then(function (response) {
        setDatabase(response.data)
     })
     .catch(function (error) {
       console.log(error);
     });
    
   }
   getData(dataForm)

 }, [dataForm])
 

  

  
  return (
    <Grid
      minH="100vh"
      h="100%"
      templateRows="repeat(1, 1fr)"
      templateColumns="repeat(5, 1fr)"
      gap={2}
    >
      <GridItem padding="5px" rowSpan={2} colSpan={1} bg="#F6F6F6">
        <Sidebar SidebarData={(e)=>{setDataForm(e)}}/>
      </GridItem>
      <GridItem h="100%" colSpan={4}>
        <HStack w="100%" h="100%" align="start">
          <Graph dataBase={database}/>
        </HStack>
      </GridItem>

      <GridItem h="70px" colSpan={4} bg="tomato">
        <HStack h="100%">{/* <FooterChart /> */}</HStack>
      </GridItem>
    </Grid>
  );
};
