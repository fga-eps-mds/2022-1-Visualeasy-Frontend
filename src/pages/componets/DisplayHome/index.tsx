import React from "react";
import { Grid, GridItem, HStack } from "@chakra-ui/react";
import Sidebar from "../Sidbar";
import Graph from "../Graph";
// import Chart from "./componetes/Charts";
// import FooterChart from "./componetes/FooterCharts";
export default function DisplayHome() {
  return (
    <Grid
      minH="100vh"
      h="100%"
      templateRows="repeat(1, 1fr)"
      templateColumns="repeat(5, 1fr)"
      gap={2}
    >
      <GridItem padding="5px" rowSpan={2} colSpan={1} bg="#F6F6F6">
        <Sidebar/>
      </GridItem>
      <GridItem h="100%" colSpan={4}>
        <HStack w="100%" h="100%" align="start">
          <Graph/>
        </HStack>
      </GridItem>

      <GridItem h="70px" colSpan={4} bg="tomato">
        <HStack h="100%">{/* <FooterChart /> */}</HStack>
      </GridItem>
    </Grid>
  );
};
