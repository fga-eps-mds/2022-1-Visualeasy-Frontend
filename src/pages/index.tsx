import DisplayHome from "./components/DisplayHome"
const Home = ({person}) => {
  return (
    <DisplayHome displayHomeData = {person}/>
  )
}

export default Home


export async function getServerSideProps(context) {
  let query = context.query

  const tidp = query.variavel
  if (query) {
  var array = tidp?.split(',').map((n)=> {
    return n;
  })
  const person =
  {
    id: query.id,
    intervalo: Number(query.intervalo),
    startDate:query.startDate,
    endDate:query.endDate,
    granularity:query.granularity,
    variavel:array,
    graphName:query.graphName
  };

  return {
    props: {person}, // will be passed to the page component as props
  }}
  return{props:{}}
}