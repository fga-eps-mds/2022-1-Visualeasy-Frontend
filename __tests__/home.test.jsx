// __tests__/index.test.jsx

import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import { ChakraProvider } from '@chakra-ui/react'
import DisplayHome from '../src/pages/index'
import {listaNomeVariaveis , postAllData as apiPost} from 'src/pages/api/api.ts'
import Graph from '../src/pages/components/Graph';
import Sidebar from '../src/pages/components/Sidebar';
import RatioSelect from '../src/pages/components/RatioTimeChart/Index';
import FormGraph from '../src/pages/components/FormShowInfo';
import MyApp from '../src/pages/_app'
import {getServerSideProps} from '../src/pages/index'
import Document from '../src/pages/_document'
import '@testing-library/jest-dom'
import { rest } from "msw"
import { setupServer } from "msw/node"
import renderer from 'react-test-renderer';

import axios from "axios";

const url = "http://localhost";

const chakraWrapper = ({ children }) => <ChakraProvider>{children}</ChakraProvider>
const DATE_TO_USE = new Date('2022-03-16T07:42:10');
const _Date = Date;
global.Date = jest.fn(() => DATE_TO_USE);
global.Date.UTC = _Date.UTC;
global.Date.parse = _Date.parse;
global.Date.now = _Date.now;

const person =
{
  id: 1,
  intervalo: 1,
  startDate:"30-04-2022",
  endDate:"30-06-2022",
  granularity:"month",
  variavel: ["Pikashu"],
  graphName:"Grafico 1"
};

const query =
{
  id: 1,
  intervalo: 1,
  startDate:"30-04-2022",
  endDate:"30-06-2022",
  granularity:"month",
  variavel: "Pikashu",
  graphName:"Grafico 1"
};

const variavelNameGet = rest.get(`${url}/variavel`, (req, res, ctx) => {
  return res(
    ctx.status(200),
    ctx.json({
      "variavels": [
        {
          "variavel": "Bulbassauro"
        },
        {
          "variavel": "Charmander"
        },
        {
          "variavel": "Pikachu"
        },
      ],
      "resposta": "Sucesso!!"
    }))
})

const postAllData = rest.post(`${url}/variavel/filtered`, (req, res, ctx) => {
  return res(
    ctx.status(200),
    ctx.json({
      "variavels": [
        {
          "date": "30-6-2022",
          "valor": "5.08"
        },
        {
          "date": "1-7-2022",
          "valor": "5.04"
        }
      ],
      "resposta": "Sucesso!!"
    })
  )
})

const postAllDataPredefinido = rest.post(`${url}/variavel/filteredByPeriod`, (req, res, ctx) => {
  return res(
    ctx.status(200),
    ctx.json({
      "variavels": [
        {
          "date": "30-6-2022",
          "valor": "5.08"
        },
        {
          "date": "1-7-2022",
          "valor": "5.04"
        }
      ],
      "resposta": "Sucesso!!"
    })
  )
})

const handlers = [variavelNameGet, postAllData, postAllDataPredefinido]

const server = new setupServer(...handlers);

beforeAll(() => server.listen({
  onUnhandleRequest: 'error'
}));
afterEach(() => server.resetHandlers());
afterAll(() => server.close())


describe('DisplayHome', () => {
  test('DisplayHome deve existir', () => {
    const displayHome = render(<DisplayHome />, {wrapper: chakraWrapper});
    expect(displayHome).not.toBeNull();
  });
});

describe('MyApp', () => {
  it('MyApp deve existir"', () => {
    const myApp =()=>{
      render(<MyApp />)
    } ;
    expect(myApp).not.toBeNull();
  });
});

describe('Document', () => {
  it('Document deve existir"', () => {
    const myApp =()=>{
      render(<Document />)
    } ;
    expect(myApp).not.toBeNull();
  });
});

describe('Logo', () => {
  test('Logo must have src = "/logo-retangular.png" and alt = "Logo"', () => {
    const sidebar = render(<Sidebar />, {wrapper: chakraWrapper});
    const logo = sidebar.container.querySelector('#logo-retangular');
    // expect(logo).toHaveAttribute('src', '/_next/static/media/logo-retangular.8228d07f.png');
    expect(logo).toHaveAttribute('alt', 'Logo');
  });
});

describe('Tabs', () => {
  test('Tabs ', async () => {
    render(<Sidebar SidebarData={person} />, {wrapper: chakraWrapper});
    const button = screen.getByText(/Gráfico 1/i, { selector: 'button' });
    expect(button).not.toBeNull();
  });
});

describe('Option', () => {
  test('Options should exists', async () => {
    const SidebarData = () => console.log("MOCK FUNÇÃO")
    render(<FormGraph FormGraphProps={(e) => { SidebarData({ ...e }) }} />, {wrapper: chakraWrapper});
    fireEvent.click(screen.getByText("Selecione as variáveis"));

    setTimeout(() => {
      fireEvent.click(screen.getAllByDisplayValue('Charmander'));
    }, 1000);

    setTimeout(() => {
      expect(widget.find('span').text().contains('Charmander').toBe(true));
    }, 1000);
  });
});

describe('Radio', () => {
  test('Radio intervalo ', async () => {
    const setGraphParameters = () => console.log("MOCK FUNÇÃO")
    render(<RatioSelect RatioRange={(e) => setGraphParameters(e)} />);
    const TitleRadio = screen.getByText(/Selecione o intervalo:/i);
    expect(TitleRadio).not.toBeNull();
    const TitleRadio1 = screen.getByText(/1 Hora/i);
    expect(TitleRadio1).not.toBeNull();
    const TitleRadio2 = screen.getByText(/1 Dia/i);
    expect(TitleRadio2).not.toBeNull();
    const TitleRadio3 = screen.getByText(/1 Sem/i);
    expect(TitleRadio3).not.toBeNull();
    const TitleRadio4 = screen.getByText(/1 Mês/i);
    expect(TitleRadio4).not.toBeNull();
    const TitleRadiop = screen.getByText(/Personalizado/i);
    expect(TitleRadiop).not.toBeNull();
  });
});

describe('Button', () => {
  test('Button gerar grafico ', async () => {
    render(<Sidebar SidebarData={person} />, {wrapper: chakraWrapper});
    const button = screen.getByText(/Gerar Gráfico/i, { selector: 'button' });
    expect(button).not.toBeNull();
  });
});

describe('closeButton', () => {
  test('CloseButton está ativado nas abas', async () => {
    render(<Sidebar SidebarData={person} />, {wrapper: chakraWrapper});
    const button = screen.findByLabelText('button', { name: /Close/i });
    expect(button).not.toBeNull();
  });
});

describe('RatioSelect', () => {
  test('Deve renderizar os campos de Início e Fim', async () => {
    const setGraphParameters = () => console.log("MOCK FUNÇÃO")
    render(<RatioSelect RatioRange={(e) => setGraphParameters(e)} />);
    const TitleRadio = screen.getByText(/Personalizado/i);
    fireEvent.click(TitleRadio);
    expect(screen.getByText('Início:')).not.toBeNull();
    expect(screen.getByText('Fim:')).not.toBeNull();
  });
});

describe('Graph Tempo Personalizado', () => {
  test('Deve renderizar o grafico com a opção de tempo personalizado', async () => {
    const dataForm = {
      variavel: ["Pikashu", "Outro"],
      intervalo: 5,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: "year"
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})


describe('Graph Tempo Predefinido', () => {
  test('Deve renderizar o grafico com a opção de tempo predefinido', async () => {
    const dataForm = {
      variavel: ["Pikashu"],
      intervalo: 2,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: "month"
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})

describe('Graph Tempo Personalizado Vazio', () => {
  test('Deve renderizar o grafico com a opção de tempo personalizado sem granularidade', async () => {
    const dataForm = {
      variavel: ["Picachu"],
      intervalo: 2,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: ""
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})

describe('Graph Tempo Personalizado Segundo', () => {
  test('Deve renderizar o grafico com a opção de tempo personalizado em segundos', async () => {
    const dataForm = {
      variavel: ["Picachu"],
      intervalo: 2,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: "second"
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})

describe('Graph Tempo Personalizado Minuto', () => {
  test('Deve renderizar o grafico com a opção de tempo personalizado em minutos', async () => {
    const dataForm = {
      variavel: ["Picachu"],
      intervalo: 2,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: "minute"
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})

describe('Graph Tempo Personalizado Dias', () => {
  test('Deve renderizar o grafico com a opção de tempo personalizado em dias', async () => {
    const dataForm = {
      variavel: ["Picachu"],
      intervalo: 2,
      startDate: "30-04-2022",
      endDate: "30-06-2022",
      granularity: "day"
    }
    const graph = renderer.create(<Graph dataForm={dataForm} />).toJSON();
    expect(graph).toMatchSnapshot();
  })
})


describe('getServerSideProps', () => {
  test('Deve retornar person', async () => {
    const context = {
      query: query
    }
    const props = getServerSideProps(context);
    expect(props).not.toBeNull();
  })
  test('Deve retornar props vazia', async () => {
    const context = {
      query: {}
    }
    const props = getServerSideProps(context);
    expect(props).not.toBeNull();
  })
})

describe('Exceptions', () => {
  test('listaNomeVariaveis Exception', async () => {
    jest.spyOn(axios, 'get').mockImplementation((req) => {
      throw "Exception Occurred";
    });
    const test = listaNomeVariaveis();
    expect(test).toStrictEqual(Promise.resolve({}));
  });
  test('apiPost Exception', async () => {
    jest.spyOn(axios, 'post').mockImplementation((req) => {
      throw "Exception Occurred";
    });
    const test = apiPost();
    expect(test).toStrictEqual(Promise.resolve({}));
  });
  test('apiPost Promise error', async () => {
    jest.spyOn(axios, 'post').mockImplementation((req) => {
      return Promise.reject();
    });
    const test = apiPost()
    expect(test).toStrictEqual(Promise.resolve({}));
  });
});