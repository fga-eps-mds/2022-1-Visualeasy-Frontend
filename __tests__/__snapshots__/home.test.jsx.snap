// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Graph Tempo Personalizado Deve renderizar o grafico com a opção de tempo personalizado 1`] = `
<div
  className="css-0"
>
  <canvas
    className="Grafico"
    height={150}
    role="img"
    width={300}
  />
  <div
    className="chakra-stack css-0"
  >
    <button
      className="css-0"
      onClick={[Function]}
      placeholder="Download"
    >
      <img
        className="chakra-image css-0"
        id="screenshot-icon"
        src="images/screenshot-icon.svg"
      />
    </button>
    <a
      download="16032022074210"
      href="data:text/csv;charset=utf-8,﻿
"
      onClick={[Function]}
      target="_self"
    >
      <button
        aria-label="download"
        className="chakra-button css-0"
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <polyline
            points="7 10 12 15 17 10"
          />
          <line
            x1="12"
            x2="12"
            y1="15"
            y2="3"
          />
        </svg>
      </button>
    </a>
    <button
      aria-label="compartilhar"
      className="chakra-button css-0"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 0 0 0-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Graph Tempo Predefinido Deve renderizar o grafico com a opção de tempo predefinido 1`] = `
<div
  className="css-0"
>
  <canvas
    className="Grafico"
    height={150}
    role="img"
    width={300}
  />
  <div
    className="chakra-stack css-0"
  >
    <button
      className="css-0"
      onClick={[Function]}
      placeholder="Download"
    >
      <img
        className="chakra-image css-0"
        id="screenshot-icon"
        src="images/screenshot-icon.svg"
      />
    </button>
    <a
      download="16032022074210"
      href="data:text/csv;charset=utf-8,﻿
"
      onClick={[Function]}
      target="_self"
    >
      <button
        aria-label="download"
        className="chakra-button css-0"
        type="button"
      >
        <svg
          aria-hidden={true}
          fill="none"
          focusable={false}
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
          />
          <polyline
            points="7 10 12 15 17 10"
          />
          <line
            x1="12"
            x2="12"
            y1="15"
            y2="3"
          />
        </svg>
      </button>
    </a>
    <button
      aria-label="compartilhar"
      className="chakra-button css-0"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M752 664c-28.5 0-54.8 10-75.4 26.7L469.4 540.8a160.68 160.68 0 0 0 0-57.6l207.2-149.9C697.2 350 723.5 360 752 360c66.2 0 120-53.8 120-120s-53.8-120-120-120-120 53.8-120 120c0 11.6 1.6 22.7 4.7 33.3L439.9 415.8C410.7 377.1 364.3 352 312 352c-88.4 0-160 71.6-160 160s71.6 160 160 160c52.3 0 98.7-25.1 127.9-63.8l196.8 142.5c-3.1 10.6-4.7 21.8-4.7 33.3 0 66.2 53.8 120 120 120s120-53.8 120-120-53.8-120-120-120zm0-476c28.7 0 52 23.3 52 52s-23.3 52-52 52-52-23.3-52-52 23.3-52 52-52zM312 600c-48.5 0-88-39.5-88-88s39.5-88 88-88 88 39.5 88 88-39.5 88-88 88zm440 236c-28.7 0-52-23.3-52-52s23.3-52 52-52 52 23.3 52 52-23.3 52-52 52z"
        />
      </svg>
    </button>
  </div>
</div>
`;
